pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -230,13 +232,15 @@ func (r *DeploymentReconciler) checkDeploymentExist(ctx context.Context, client
 		log.Error(err, "Failed to perform dry-run update of deployment", "Deployment", deployment.Name)
 		return constants.CheckResultUnknown, nil, err
 	}
+
 	if diff, err := kmp.SafeDiff(deployment.Spec, existingDeployment.Spec, ignoreFields); err != nil {
+		log.Error(err, "Failed to diff deployments", "Deployment", deployment.Name)
```

Comment:
Is this needed? err will be returned by the function so maybe it will be printed there